tests/algorithms/test_algorithm_resumption.py

```diff
@@ -0,0 +1,236 @@
+    composer_trainer_hparams.train_subset_num_batches = 5
+    composer_trainer_hparams.eval_subset_num_batches = 5
+    composer_trainer_hparams.device = device_hparams
+    if deepspeed_enabled:
+        assert zero_stage is not None
+        if zero_stage > 0:
+            composer_trainer_hparams.deterministic_mode = False
+            if model_name is not None:
+                pytest.skip(
+                    textwrap.dedent(f"""\
+                        Skipping test since deterministic mode is required for
+                        non-trivial models, but deterministic mode isn't compatible with deepspeed
+                        zero stage {zero_stage}"""))
+        composer_trainer_hparams.deepspeed = {"zero_optimization": {"stage": zero_stage}}
+
+    checkpoint_a_folder = "first"
+    composer_trainer_hparams.save_folder = checkpoint_a_folder
+    composer_trainer_hparams.save_interval = save_interval
+    composer_trainer_hparams.seed = seed
+
+    composer_trainer_hparams.validate_every_n_batches = 1 if resume_file.startswith("ba") else 0
+    composer_trainer_hparams.validate_every_n_epochs = 1 if resume_file.startswith("ep") else 0
+    first_trainer = composer_trainer_hparams.initialize_object()
+    first_trainer.fit()
+    save_interval_time = Time.from_timestring(save_interval)
+    if save_interval_time.unit == TimeUnit.EPOCH:
+        expected_num_checkpoints = ((num_epochs - 1) // save_interval_time.value) + 1
+    else:
+        expected_num_checkpoints = (
+            (composer_trainer_hparams.train_subset_num_batches * num_epochs - 1) // save_interval_time.value) + 1
+    checkpoint_saver = None
+    for callback in first_trainer.state.callbacks:
+        if isinstance(callback, CheckpointSaver):
+            checkpoint_saver = callback
+    assert checkpoint_saver is not None
+    assert len(checkpoint_saver.saved_checkpoints) == expected_num_checkpoints
+    checkpoint_a_file_path = os.path.join(checkpoint_a_folder, resume_file)
+    checkpoint_b_file_path = os.path.join(run_directory.get_node_run_directory(), "rank_{rank}", checkpoint_a_folder,
+                                          final_checkpoint)
+
+    second_trainer_hparams = TrainerHparams.create(data=composer_trainer_hparams.to_dict(), cli_args=False)
+    checkpoint_b_folder = "second"
+
+    second_trainer_hparams.save_folder = checkpoint_b_folder
+    second_trainer_filepath = os.path.join(run_directory.get_node_run_directory(), "rank_{rank}",
+                                           checkpoint_a_file_path)
+    second_trainer_hparams.load_path_format = second_trainer_filepath
+    second_trainer_hparams.load_weights_only = False
+    second_trainer_hparams.load_strict_model_weights = False
+
+    second_trainer = second_trainer_hparams.initialize_object()
+    second_trainer.fit()
+    checkpoint_c_file_path = os.path.join(run_directory.get_node_run_directory(), "rank_{rank}", checkpoint_b_folder,
```

Comment:
I could be mistaken, but I thought this creates a second directory so that the second (resumed) training run doesn't overwrite the checkpoints from the first run.